### PR TITLE
Fix MetadataTest failures

### PR DESCRIPTION
**Pull request by dreab8**
- the behaviour introduced by HHH-4702 has been reverted, SingularAttribute#getBindableJavaType() and SingularAttribute#getType().getJavaType() return Integer.class also for int attrivutes
- IdClass, subclass should have the same id attributes declared on the superclass, this fix the MetadaTest#testEntity testing

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metadata/MetadataTest.java [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+    @Test
+    public void testBasic(EntityManagerFactoryScope scope) {
+        final EntityType<Fridge> entityType = scope.getEntityManagerFactory().getMetamodel().entity(Fridge.class);
+        final SingularAttribute<? super Fridge, Integer> singularAttribute = entityType.getDeclaredSingularAttribute(
+                "temperature",
+                Integer.class
+        );
+		assertEquals( Integer.class, singularAttribute.getBindableJavaType() );
+		assertEquals( Integer.class, singularAttribute.getType().getJavaType() );
+//        assertEquals(int.class, singularAttribute.getBindableJavaType());
```
Comment (sebersole) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metadata/MetadataTest.java:231]: Can we remove these commented-out lines so we don't confused ourselves again? ;)

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metadata/MetadataTest.java [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+        assertFalse(singularAttribute.isOptional());
+        assertFalse(entityType.getDeclaredSingularAttribute("brand", String.class).isOptional());
+        assertEquals(Type.PersistenceType.BASIC, singularAttribute.getType().getPersistenceType());
+        final Attribute<? super Fridge, ?> attribute = entityType.getDeclaredAttribute("temperature");
+        assertNotNull(attribute);
+        assertEquals("temperature", attribute.getName());
+        assertEquals(Fridge.class, attribute.getDeclaringType().getJavaType());
+        assertEquals(Attribute.PersistentAttributeType.BASIC, attribute.getPersistentAttributeType());
+		assertEquals( Integer.class, attribute.getJavaType() );
+//        assertEquals(int.class, attribute.getJavaType());
```
Comment (sebersole) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metadata/MetadataTest.java:244]: same here

**Review by sebersole — APPROVED** [review]
Minor comments.  Other than that, LGTM

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metadata/MetadataTest.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metadata/MetadataTest.java]: I'm going to remove them, thanks @sebersole !

**Pull request merged**